# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../etcnet/0004-do-not-add-dev-xxx-suffix-when-handling-special-route-types.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From ed74dc8cb0ba0ecd95c75b57e1907ab8c44bb5ba Mon Sep 17 00:00:00 2001
From: Sergey Bolshakov <sbolshakov@altlinux.org>
Date: Sun, 17 Jan 2010 21:18:17 +0300
Subject: [PATCH 4/4] do not add 'dev xxx' suffix when handling special route types

---
 etc/net/scripts/functions-ipv4 |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/etc/net/scripts/functions-ipv4 b/etc/net/scripts/functions-ipv4
index 5379c68..51625ab 100644
--- a/etc/net/scripts/functions-ipv4
+++ b/etc/net/scripts/functions-ipv4
@@ -157,7 +157,9 @@ config_ipv4_routes_rules()
 		while read FIRST REST; do
 			# If 'dev' is not specified, add current dev.
 			SUFFIX=''
-			echo "$FIRST $REST" | egrep -q '[[:space:]]dev[[:space:]]' || SUFFIX="dev $NAME"
+			if ! echo "$FIRST $REST" | egrep -q '(^|[[:space:]])(throw|unreachable|prohibit|blackhole)[[:space:]]'; then
+				echo "$FIRST $REST" | egrep -q '[[:space:]]dev[[:space:]]' || SUFFIX="dev $NAME"
+			fi
 			# If operation is not specified, use 'append'.
 			case "$FIRST" in
 				add|change|append|replace)
-- 
1.6.6.2

